src/core/common/message.cpp

```diff
@@ -977,16 +972,15 @@ otError PriorityQueue::Enqueue(Message &aMessage)
     mTails[priority] = &aMessage;
 
 exit:
-    return error;
+    return;
 }
 
-otError PriorityQueue::Dequeue(Message &aMessage)
+void PriorityQueue::Dequeue(Message &aMessage)
 {
-    otError  error = OT_ERROR_NONE;
     uint8_t  priority;
     Message *tail;
 
-    VerifyOrExit(aMessage.GetPriorityQueue() == this, error = OT_ERROR_NOT_FOUND);
+    VerifyOrExit(aMessage.GetPriorityQueue() == this, OT_NOOP);
```

Comment:
```suggestion
    VerifyOrExit(aMessage.IsInAQueue(), OT_NOOP);
    
    OT_ASSERT(aMessage.GetPriorityQueue() == this);
```
Just let the cases that the message is in another queue to fail.